### ISSUE TITLE
Tasks cycle again after completing assigned tasks

### ISSUE THREAD

**Issue by Onibus** [nvll/xivtools#26]
OS: Windows
Build: a5edfc62d6f4d22645fd5553781a9253027dc48d

When tasks are assigned, they are completed as expected. The problem is that once all tasks have been completed, talan.exe will cycle through the task list again. 

Below is the log information. I assigned a task to craft two maple lumber. At `12:23:57,265`, we see the Talan GUI appear for a split second before starting the previously completed task again.
```
2019-08-11 12:22:54,529 [36mINFO [0m [winit::platform::platform::window] Guessed window DPI factor: 1
2019-08-11 12:23:14,564 [35mDEBUG[0m [xiv::ui] clearing the game window
2019-08-11 12:23:22,936 [36mINFO [0m [talan::craft] changing to gearset 4
2019-08-11 12:23:26,607 [36mINFO [0m [talan::craft] selecting recipe...
2019-08-11 12:23:29,725 [36mINFO [0m [talan::craft] crafting Maple Lumber 1/2
2019-08-11 12:23:33,038 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (101.4778ms)
2019-08-11 12:23:35,740 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (2.7020104s)
2019-08-11 12:23:41,442 [36mINFO [0m [talan::craft] crafting Maple Lumber 2/2
2019-08-11 12:23:44,754 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (101.0794ms)
2019-08-11 12:23:47,456 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (2.7019066s)
2019-08-11 12:23:57,265 [36mINFO [0m [winit::platform::platform::window] Guessed window DPI factor: 1
2019-08-11 12:23:57,369 [35mDEBUG[0m [xiv::ui] clearing the game window
2019-08-11 12:24:05,742 [36mINFO [0m [talan::craft] changing to gearset 4
2019-08-11 12:24:09,408 [36mINFO [0m [talan::craft] selecting recipe...
2019-08-11 12:24:12,525 [36mINFO [0m [talan::craft] crafting Maple Lumber 1/2
2019-08-11 12:24:15,838 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (101.2432ms)
2019-08-11 12:24:18,540 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (2.7020952s)
```

**Comment by Onibus** [comment]
I added a few debug log lines to try and narrow down where the issue might be. It appears to be something with the loop in main.rs. 

Once all tasks are finished, the loop in https://github.com/nvll/xivtools/blob/a5edfc62d6f4d22645fd5553781a9253027dc48d/talan/src/main.rs#L81 completes as intended (added `log::debug!("main.rs -> loop start");` and  at beginning of `loop{}` and `log::debug!("main.rs -> loop end after craft_items finished");` at end of `loop{}`). Immediately after, it feels as if https://github.com/nvll/xivtools/blob/a5edfc62d6f4d22645fd5553781a9253027dc48d/talan/src/main.rs#L82 is not being honored to break out of the loop.
```
2019-08-11 13:37:33,881 [35mDEBUG[0m [talan::gui] mod.rs -> State.exit_gui = true
2019-08-11 13:37:33,884 [35mDEBUG[0m [xiv::ui] clearing the game window
2019-08-11 13:37:42,261 [36mINFO [0m [talan::craft] changing to gearset 4
2019-08-11 13:37:45,937 [36mINFO [0m [talan::craft] selecting recipe...
2019-08-11 13:37:49,054 [36mINFO [0m [talan::craft] crafting Maple Lumber 1/1
2019-08-11 13:37:51,256 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (101.6263ms)
2019-08-11 13:37:53,958 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (2.7022466s)
2019-08-11 13:38:01,661 [35mDEBUG[0m [talan::craft] craft.rs -> End of 'execute_task' function
2019-08-11 13:38:03,763 [35mDEBUG[0m [talan] main.rs -> loop end after craft_items finished
2019-08-11 13:38:03,763 [35mDEBUG[0m [talan] main.rs -> loop start
2019-08-11 13:38:03,770 [36mINFO [0m [winit::platform::platform::window] Guessed window DPI factor: 1
2019-08-11 13:38:03,882 [35mDEBUG[0m [xiv::ui] clearing the game window
```

If I set a `break;` at the end of `loop{}`, it correctly gets out of the loop (added `log::debug!("main.rs -> exiting fn:main");` above https://github.com/nvll/xivtools/blob/a5edfc62d6f4d22645fd5553781a9253027dc48d/talan/src/main.rs#L90)
```
2019-08-11 13:40:35,809 [35mDEBUG[0m [talan] main.rs -> loop start
2019-08-11 13:40:35,818 [36mINFO [0m [winit::platform::platform::window] Guessed window DPI factor: 1
2019-08-11 13:40:39,754 [35mDEBUG[0m [talan::gui] mod.rs -> State.exit_gui = true
2019-08-11 13:40:39,758 [35mDEBUG[0m [xiv::ui] clearing the game window
2019-08-11 13:40:48,129 [36mINFO [0m [talan::craft] changing to gearset 4
2019-08-11 13:40:51,800 [36mINFO [0m [talan::craft] selecting recipe...
2019-08-11 13:40:54,920 [36mINFO [0m [talan::craft] crafting Maple Lumber 1/1
2019-08-11 13:40:57,126 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (104.2581ms)
2019-08-11 13:40:59,827 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (2.7015486s)
2019-08-11 13:41:07,530 [35mDEBUG[0m [talan::craft] craft.rs -> End of 'execute_task' function
2019-08-11 13:41:09,632 [36mINFO [0m [talan::craft] changing to gearset 21
2019-08-11 13:41:13,351 [36mINFO [0m [talan::craft] selecting recipe...
2019-08-11 13:41:16,671 [36mINFO [0m [talan::craft] crafting Iron Ingot 1/1
2019-08-11 13:41:18,874 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (101.6338ms)
2019-08-11 13:41:21,575 [35mDEBUG[0m [talan::craft] action: careful synthesis iii (2.701332s)
2019-08-11 13:41:29,278 [35mDEBUG[0m [talan::craft] craft.rs -> End of 'execute_task' function
2019-08-11 13:41:31,380 [35mDEBUG[0m [talan] main.rs -> loop end after craft_items finished
exiting...
2019-08-11 13:41:31,381 [35mDEBUG[0m [talan] main.rs -> exiting fn:main
```

**Comment by nvll** [comment]
Addressed in 81563551d9e33876514823775a40b8f369e328cd, try using the dev branch until I merge it into master tonight.

**Comment by nvll** [comment]
Fixed in master.
